File: src-tauri/src/models/encrypted_vault_item.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
pub struct EncryptedVaultItem {
    pub name: String,
    pub username: String,
    pub password: Vec<u8>,
    pub urls: Vec<String>,
    pub notes: String,
    pub is_favourite: bool,
    pub category: String,

    pub id: String,

    pub salt: [u8; 16],
    pub nonce: [u8; 12],
}

File: src/types/encryptedVaultItem.ts
import { VaultItemCategory } from "../enums/vaultItemCategory";

interface EncryptedVaultItem {
    id: string;
    name: string;
    username: string;
    password: number[];
    urls: string[];
    notes: string;

    is_favourite: boolean;
    category: VaultItemCategory;

    salt: number[];
    nonce: number[];
}

export type { EncryptedVaultItem }